docs/docs/guides/build/external-pipelines/scalaspark-pipeline.md

```diff
@@ -0,0 +1,145 @@
+│       └── Example.scala
+│   └── build/libs
+│       └── external_scala-all.jar
+```
+
+- `dagster_code.py` will contain the Dagster orchestration code.
+- `Example.scala` will contain the Spark code to calculate pi, as well as the code to send message to Dagster.
+- `gradlew` is the gradle wrapper (a script that downloads gradle on demand - this is the recommended way to use gradle).
+- `build.gradle` contains all the build instructions to build and package `Example.scala` into a suitable jar for submission to Spark.
+- `build/libs/external_scala-all.jar` is the output of this build process (created automatically by `docker build`).
+- `Dockerfile` and `docker-compose.yml` allow for running this example in a reproducible environment using `docker compose`.
+
+## Step 1: Writing the Dagster orchestration code (dagster_code.py)
+
+We will set up a few non-default Pipes components to streamline the otherwise challenging problem of orchestrating Spark jobs.
+
+<CodeExample path="docs_snippets/docs_snippets/guides/dagster/dagster_pipes/scalaspark/dagster_code.py">
+
+* Notice that we are using S3 to pass Pipes messages from the Spark job to Dagster - so we create `PipesS3MessageReader` and `PipesS3ContextInjector` objects. (Technically, it's not strictly required to use S3 for passing the Dagster context, but storing it there will decrease the CLI arguments size).
+
+* Notice we are using CLI arguments to pass the bootstrap information from Dagster to the Spark job. We will fetch them from the `session.get_bootstrap_cli_arguments` method. We pass these arguments to `spark-submit`.
+
+:::note
+
+In other Pipes workflows, passing the bootstrap information from Dagster to the remote Pipes session is typically done via environment variables, but setting environment variables for Spark jobs can be complicated (the exact way of doing this depends on the Spark deployment) or not possible at all. CLI arguments are a convenient alternative.
+
+:::
+
+## Step 2: Use Pipes in the Spark job (Example.scala)
+
+Our example Spark workload (based on the [official example](https://github.com/apache/spark/blob/master/examples/src/main/scala/org/apache/spark/examples/SparkPi.scala)) will use Spark to calculate pi, and report the result back to Dagster.
+
+First, create a new file name `Example.scala` in the `src/main/scala/org/examples` directory, then add the following code to create a context that can be used to send messages to Dagster:
+
+<CodeExample path="docs_snippets/docs_snippets/guides/dagster/dagster_pipes/scalaspark/Example.scala" />
+
+* Note how `PipesCliArgsParamsLoader` is used to load the CLI arguments passed by Dagster. This information will be used to automatically configure `PipesS3MessageWriter` and `PipesS3ContextLoader`.
+
+* We are using the [Java Pipes Framework](https://github.com/dagster-io/community-integrations/tree/main/libraries/pipes/implementations/java) in our code (since Scala is fully compatible with Java libraries).
```

Comment:
```suggestion
* We are using the [Java Pipes Framework](https://github.com/dagster-io/community-integrations/tree/main/libraries/pipes/implementations/java) in our code, since Scala is fully compatible with Java libraries.
```